publication-commons/src/test/java/no/unit/nva/publication/permission/strategy/ContributorPermissionStrategyTest.java

```diff
@@ -0,0 +1,79 @@
+package no.unit.nva.publication.permission.strategy;
+
+import static no.unit.nva.model.PublicationOperation.UNPUBLISH;
+import static no.unit.nva.testutils.RandomDataGenerator.randomString;
+import static no.unit.nva.testutils.RandomDataGenerator.randomUri;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import no.unit.nva.model.PublicationOperation;
+import no.unit.nva.model.associatedartifacts.AssociatedArtifactList;
+import no.unit.nva.model.role.Role;
+import no.unit.nva.publication.RequestUtil;
+import nva.commons.apigateway.exceptions.UnauthorizedException;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.junit.jupiter.params.provider.EnumSource.Mode;
+
+class ContributorPermissionStrategyTest extends PublicationPermissionStrategyTest {
+
+    @ParameterizedTest(name = "Should allow verified contributor {0} operation on non-degree resources")
+    @EnumSource(value = PublicationOperation.class, mode = Mode.EXCLUDE, names = {"DELETE", "TERMINATE",
+        "TICKET_PUBLISH"})
+    void shouldAllowVerifiedContributorOnNonDegree(PublicationOperation operation)
+        throws JsonProcessingException, UnauthorizedException {
+
+        var institution = randomUri();
+        var contributor = randomString();
+        var cristinId = randomUri();
+
+        var requestInfo = createUserRequestInfo(contributor, institution, cristinId, randomUri());
+        var publication = createPublicationWithContributor(contributor, cristinId, Role.CREATOR,
+                                                           randomUri(), randomUri());
+        publication.setAssociatedArtifacts(new AssociatedArtifactList());
```

Comment:
Was wondering why this line had to be included for the test to pass, while the next test did not. In my eyes these two tests should've/could've had the same body. But it turns out that `randomPublicationNonDegree()` in `createPublicationWithContributor(...)` creates a publication with a published file, which is quite essential information in this test, as half of the UNPUBLISH-test relies on the publication containing a publish file or not. Might want to specify why this line had to be added.

Reply:
agree! will rewrite a bit!